regexec.c

```diff
@@ -3728,10 +3731,11 @@ Perl_regexec_flags(pTHX_ REGEXP * const rx, char *stringarg, char *strend,
              * Let @-, @+, $^N know */
             prog->lastparen = prog->lastcloseparen = 0;
             RXp_MATCH_UTF8_set(prog, utf8_target);
-            prog->offs[0].start = s - strbeg;
-            prog->offs[0].end = utf8_target
+            SSize_t match_start = s - strbeg;
+            SSize_t match_end = utf8_target
                 ? (char*)utf8_hop_forward((U8*)s, prog->minlenret, (U8 *) strend) - strbeg
                 : s - strbeg + prog->minlenret;
+            CLOSE_ANY_CAPTURE(prog, 0, match_start, match_end);
```

Comment:
Hrm. Ill take a look.

Reply:
I dont get it. This is added in ddeba638cfb50b03af2a8d68f2573a2fb6c18e64 and is the *entire* reason for the patch.  This is the only place in the code which does a CLOSE_CAPTURE() like behavior which does not use CLOSE_CAPTURE(). Because of when it executes it cannot use CLOSE_CAPTURE() as it does not have some of debug data available to implement that. Thus this patch splits CLOSE_CAPTURE() into CLOSE_ANY_CAPTURE() and CLOSE_CAPTURE() as a wrapper.

The whole point is harden all the places in the code which close a capture and which might operate on offs[i].start and offs[i].end, which will be replaced in a follow up patch.

So I don't get it, and I don't agree. If you want I can add more to the commit message, but this line is the entire reason the code was changed. So that I could change the definition of CLOSE_ANY_CAPTURE() and fix all the places that try to close capture buffers.

Reply:
I have updated the commit message. Also I have pushed this commit as part of https://github.com/Perl/perl5/pull/20703. I think it would be better to provide any further feedback you have there.